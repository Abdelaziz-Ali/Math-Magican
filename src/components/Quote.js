import { useState, useEffect } from 'react';
import { PropTypes } from 'prop-types';

function Quote({ sav }) {
  const [data, setData] = useState('loading...');

  useEffect(() => {
    const getQuotes = async () => {
      const myHeaders = new Headers();
      myHeaders.append('X-Api-Key', 'JApZ9IkdEvyK/Wn4uTRNkQ==vEuwff2CfhjGsGLJ');

      const requestOptions = {
        method: 'GET',
        headers: myHeaders,
        redirect: 'follow',
      };

      await fetch(
        'https://api.api-ninjas.com/v1/quotes?category=learning',
        requestOptions,
      )
        .then((response) => response.json())
        .then((result) => setData(result[0].quote))
        .catch(() => setData('Error loading Quotes!!'));
    };
    getQuotes();
  }, []);

  return (
    <>
      <p className={sav}>
        Quote :
        {data}
      </p>
    </>
  );
}
Quote.propTypes = { sav: PropTypes.string.isRequired };

export default Quote;
