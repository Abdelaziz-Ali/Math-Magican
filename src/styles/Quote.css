.quote {
  font-size: 2rem;
  font-weight: 500;
  font-style: italic;
  margin: 1rem 0;
  text-align: center;
  padding: 0 1rem;
  display: flex;
  align-content: center;
  flex-wrap: wrap;
  height: 82vh;
  justify-content: center;
  align-items: center;
}

@media screen and (max-width: 768px) {
  .quote {
    font-size: 1.3rem;
  }
}
