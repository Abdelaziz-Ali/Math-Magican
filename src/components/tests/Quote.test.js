import { expect } from '@jest/globals';
import renderer from 'react-test-renderer';
import Quote from '../Quote';

it('renders Api component', () => {
  const component = renderer.create(<Quote />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
