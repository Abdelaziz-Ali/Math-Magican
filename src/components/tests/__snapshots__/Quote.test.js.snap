// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Api component 1`] = `
<p
  className="undefined quote"
>
  Quote :
   
  loading...
  <br />
</p>
`;
